sc/social/like/utils.py

```diff
@@ -3,71 +3,55 @@
 from Products.Archetypes.interfaces import IBaseContent
 from Products.CMFPlone.utils import safe_hasattr
 from urlparse import urlparse
-from zope.annotation.interfaces import IAnnotations
-from zope.globalrequest import getRequest
 from zope.interface import Invalid
 
 
 def get_images_view(context):
-    request = getRequest()
-    key = 'cache-view-' + str(context)
-    cache = IAnnotations(request)
-    value = cache.get(key, None)
-    if not value:
-        view = context.unrestrictedTraverse('@@images', None)
-        field = 'image'
-        if view:
-            fields = ['image', 'leadImage', 'portrait']
-            if IBaseContent.providedBy(context):
-                schema = context.Schema()
-                field = [f for f in schema.keys() if f in fields]
-                if field:
-                    field = field[0]
-                    # if a content has an image field that isn't an ImageField
-                    # (for example a relation field), set field='' to avoid errors
-                    if schema[field].type not in ['image', 'blob']:
-                        field = ''
-        value = (view, field) if (view and field) else (None, None)
-        cache[key] = value
-    return value
+    view = context.unrestrictedTraverse('@@images', None)
+    field = 'image'
+    if view:
+        fields = ['image', 'leadImage', 'portrait']
+        if IBaseContent.providedBy(context):
+            schema = context.Schema()
+            field = [f for f in schema.keys() if f in fields]
+            if field:
+                field = field[0]
+                # if a content has an image field that isn't an ImageField
+                # (for example a relation field), set field='' to avoid errors
+                if schema[field].type not in ['image', 'blob']:
+                    field = ''
+    return (view, field) if (view and field) else (None, None)
 
 
 def get_content_image(context,
                       scale='large',
                       width=None,
                       height=None):
-    request = getRequest()
-    modification = context.ModificationDate()
-    key = 'cache-{0}-{1}-{2}-{3}-{4}'.format(
-        context, modification, scale, width, height)
-    cache = IAnnotations(request)
-    img = cache.get(key, None)
-    if not img:
-        view, field = get_images_view(context)
-        if view:
+    view, field = get_images_view(context)
+    img = None
```

Comment:
@hvelarde are you considering when view variable is None? I swear to you I did this because there was an exception without it, and this fixes the exception. you can check in the logs.

Also I don't want to move too much code in this PR, we have other issue for this.